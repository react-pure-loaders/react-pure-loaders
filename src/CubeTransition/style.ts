import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const amount = '50px';
const size = '10px';

const ballBeat = keyframes({
  '100%': {
    transform: 'rotate(-360deg)',
  },
  '25%': {
    transform: `translateX(${amount}) scale(0.5) rotate(-90deg)`,
  },
  '50%': {
    transform: `translate(${amount}, ${amount}) rotate(-180deg)`,
  },
  '75%': {
    transform: `translateY(${amount}) scale(0.5) rotate(-270deg)`,
  },
});

export default (color?: string) => ({
  position: 'relative' as 'relative',
  transform: `translate(-${parseInt(amount, 10) / 2}, -${parseInt(amount, 10) / 2})`,
  ' > div': {
    ...globalAnimation(),
    animation: `${ballBeat} 1.6s 0s infinite ease-in-out`,
    backgroundColor: color || PRIMARY_COLOR,
    height: size,
    left: '-5px',
    position: 'absolute' as 'absolute',
    top: '-5px',
    width: size,

    '&:last-child': {
      animationDelay: '-0.8s',
    },
  },
});
