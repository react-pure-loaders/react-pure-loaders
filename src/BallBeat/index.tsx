/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
