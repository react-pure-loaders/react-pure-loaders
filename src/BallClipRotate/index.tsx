/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
