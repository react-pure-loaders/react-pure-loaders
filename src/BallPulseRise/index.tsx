/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
