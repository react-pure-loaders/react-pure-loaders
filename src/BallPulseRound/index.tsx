/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
