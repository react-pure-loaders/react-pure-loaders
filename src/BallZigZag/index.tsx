/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
