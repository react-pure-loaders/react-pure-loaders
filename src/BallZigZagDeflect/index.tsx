/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
