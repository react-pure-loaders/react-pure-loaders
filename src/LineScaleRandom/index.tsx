/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
