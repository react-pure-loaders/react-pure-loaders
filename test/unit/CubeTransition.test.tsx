import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import CubeTransition from '../../src/CubeTransition';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<CubeTransition>', () => {
    afterEach(cleanup);

    test('CubeTransition should match snapshot', () => {
        const { container } = render(<CubeTransition loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('CubeTransition should have default color', () => {
        const { container } = render(<CubeTransition loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('CubeTransition should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<CubeTransition color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('CubeTransition should have no children', () => {
        const { container } = render(<CubeTransition loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('CubeTransition should have three children', () => {
        const { container } = render(<CubeTransition loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(3);
    });
});
