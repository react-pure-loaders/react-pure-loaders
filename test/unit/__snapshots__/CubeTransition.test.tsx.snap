// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CubeTransition> CubeTransition should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: rotate(-360deg);
    -moz-transform: rotate(-360deg);
    -ms-transform: rotate(-360deg);
    transform: rotate(-360deg);
  }

  25% {
    -webkit-transform: translateX(50px) scale(0.5) rotate(-90deg);
    -moz-transform: translateX(50px) scale(0.5) rotate(-90deg);
    -ms-transform: translateX(50px) scale(0.5) rotate(-90deg);
    transform: translateX(50px) scale(0.5) rotate(-90deg);
  }

  50% {
    -webkit-transform: translate(50px, 50px) rotate(-180deg);
    -moz-transform: translate(50px, 50px) rotate(-180deg);
    -ms-transform: translate(50px, 50px) rotate(-180deg);
    transform: translate(50px, 50px) rotate(-180deg);
  }

  75% {
    -webkit-transform: translateY(50px) scale(0.5) rotate(-270deg);
    -moz-transform: translateY(50px) scale(0.5) rotate(-270deg);
    -ms-transform: translateY(50px) scale(0.5) rotate(-270deg);
    transform: translateY(50px) scale(0.5) rotate(-270deg);
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translate(-25, -25);
  -moz-transform: translate(-25, -25);
  -ms-transform: translate(-25, -25);
  transform: translate(-25, -25);
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1.6s 0s infinite ease-in-out;
  animation: animation-0 1.6s 0s infinite ease-in-out;
  background-color: #ffffff;
  height: 10px;
  left: -5px;
  position: absolute;
  top: -5px;
  width: 10px;
}

.emotion-0>div:last-child {
  -webkit-animation-delay: -0.8s;
  animation-delay: -0.8s;
}

<div
  class="emotion-0"
>
  <div />
  <div />
</div>
`;
